Framework/PythonInterface/mantid/plots/mantidaxes.py

```diff
@@ -1222,8 +1225,13 @@ def set_xscale(self, *args, **kwargs):
             self.minorticks_on()
         else:
             self.minorticks_off()
+        if is_waterfall:
+            self.set_waterfall(True)
```

Comment:
Just a couple of things about these calls to toggle set_waterfall off\on. They apply to set_yscale as well. If I'm reading this correctly, the calls seem to provoke the waterfall offset calculation into resetting when someone changes the axis scale from log to linear or v.v:
- I don't think it matters where the calls to set_waterfall(False) and set_waterfall(True) happen as long as they occur in that order. I'm wondering if you can do the following. This would make it more clear what's happening and would remove the suggestion that something in L1220-1227 requires the waterfall status to be False:
```
if is_waterfall:
    self.set_waterfall(False)
    self.set_waterfall(True)
```
- if the user has specified a custom offset or a fill using the buttons in my screenshot below these get dropped by these calls to set_waterfall when you change an axis from linear to log. There are three further parameters in set_waterfall (x_offset, y_offset and fill). If you supply values for these in the True call perhaps the previous user settings will be kept:
![image](https://user-images.githubusercontent.com/56295817/172808443-6d539419-d087-4619-889c-2abec7b96669.png)

Reply:
The set waterfall does need to be done seperatly, the first call sets all the offsets to 0, and the second sets it to the default offset for the scale the axes is in.

Reply:
Yes makes sense for there to be 2 calls to set_waterfall. My point was whether they need to be separated out on different sides of the call to set_xscale or whether they can be done on consecutive lines. At the moment they're separated which makes it look like set_xscale requires the waterfall to be False\off so might be better if they were on consecutive lines. Bit of a pedantic point to improve the readability so fine to leave. 
Separate to this please can you make a change to get the fill setting to be retained - similar to what you did on the earlier commit to keep the previous x\y offsets